===== Patch instructions generated by SAVER =====
[+] { Insert: if (true) free(*(getstr:ret)) at 35 (line 2029, column 11),
        Insert: if ({ getstr:n$15 == { 0 } }) free(*(getstr:buf)) at 18 (line 2044, column 10) }
===== Actual patch in diff format =====
--- a/table1/benchmarks/inetutils-1.9.4/src/rshd.c
+++ b/table1/benchmarks/inetutils-1.9.4/src/rshd.c
@@ -2028,6 +2028,7 @@ getstr (const char *err)
 
       if (rd <= 0)
 	{
+		free(buf);
 	  if (rd == 0)
 	    rshd_error ("EOF reading %s\n", err);
 	  else
@@ -2041,7 +2042,9 @@ getstr (const char *err)
 	  /* Not very much room left in our buffer, grow it. */
 	  size_t end_offs = end - buf;
 	  buf_len += buf_len;
-	  buf = realloc (buf, buf_len);
+		tmp = realloc (buf, buf_len);
+		if (tmp == NULL)
+			free(buf);
 	  if (!buf)
 	    {
 	      rshd_error ("Out of space reading %s\n", err);
